test/ee2_SDKMethodRunner_test.py

```diff
@@ -1461,3 +1471,62 @@ def test_check_jobs_date_range(self, condor_mock):
             for key in job_state_limit_desc.keys():
                 print(key)
                 print(job_state_limit_desc[key])
+
+    # The Saga Continues
+    @patch("lib.execution_engine2.utils.Condor.Condor", autospec=True)
+    def test_check_jobs_date_range_freetext(self, condor_mock):
+
+        user_name = "wsadmin"
+
+        runner = self.getRunner()
+        runner.workspace_auth = MagicMock()
+        runner.auth.get_user = MagicMock(return_value=user_name)
+        runner.is_admin = True
+        runner._is_admin = MagicMock(return_value=True)
+
+        runner.workspace_auth.can_read = MagicMock(return_value=True)
+        runner.get_permissions_for_workspace = MagicMock(return_value=True)
+        runner._get_module_git_commit = MagicMock(return_value="git_commit_goes_here")
+        runner.get_condor = MagicMock(return_value=condor_mock)
+        # ctx = {"user_id": self.user_id, "wsid": self.ws_id, "token": self.token}
+        job = get_example_job().to_mongo().to_dict()
+        job["method"] = job["job_input"]["app_id"]
+        job["app_id"] = job["job_input"]["app_id"]
+        job["app_id"] = "super_module.weak_weak_weak"
+        job["msg"] = "strong"
+
+        job2 = get_example_job().to_mongo().to_dict()
+        job2["method"] = job2["job_input"]["app_id"]
+        job2["app_id"] = job2["job_input"]["app_id"]
+        job2["app_id"] = "super_module.weak_function"
+        job2["msg"] = "weak"
+
+        si = submission_info(clusterid="test", submit=job, error=None)
+        condor_mock.run_job = MagicMock(return_value=si)
+        runner._get_client_groups = MagicMock(return_value={})
+
+        print("About to submit job with ", job["app_id"])
+        job_id1 = runner.run_job(params=job)
+
+        print("About to submit job with ", job2["app_id"])
+        job_id2 = runner.run_job(params=job2)
+
+        now = datetime.utcnow()
+        yesterday = now - timedelta(days=1)
+        tomorrow = now + timedelta(days=1)
+
+        job_states = runner.check_jobs_date_range_for_user(
+            creation_start_time=str(yesterday),
+            creation_end_time=str(tomorrow),
+            user=user_name,
+            freetext_search="weak_weak_weak",
+        )
+        print(job_states)
+        self.assertEquals(job_states["jobs"][0]["_id"], job_id1)
+
+        # So "weak" is not a match, but "weak_weak_weak" is a match
+        # So maybe we are better off with doing a filter on the jobs via iterating over the job_states object and if so, keep the record, if not, ignore it
```

Comment:
I agree. I think we should just iterate over jobs. Otherwise, I am sure it will report back to us.